candidate/views_admin.py

```diff
@@ -903,19 +905,361 @@ def candidate_retrieve_photos_view(request, candidate_id):
+        return HttpResponseRedirect(reverse('candidate:candidate_list', args=()) +
+                                    '?google_civic_election_id=' + str(google_civic_election_id) +
+                                    '&state_code=' + str(state_code))
+
+    candidate2_results = candidate_campaign_manager.retrieve_candidate_campaign_from_we_vote_id(candidate2_we_vote_id)
+    if candidate2_results['candidate_campaign_found']:
+        candidate2_on_stage = candidate2_results['candidate_campaign']
+    else:
+        messages.add_message(request, messages.ERROR, 'Could not retrieve candidate 2.')
+        return HttpResponseRedirect(reverse('candidate:candidate_list', args=()) +
+                                    '?google_civic_election_id=' + str(google_civic_election_id) +
+                                    '&state_code=' + str(state_code))
+
+    # TODO: Migrate images
+
+    # Merge politician data
+    politician1_we_vote_id = candidate1_on_stage.politician_we_vote_id
+    politician2_we_vote_id = candidate2_on_stage.politician_we_vote_id
+    if positive_value_exists(politician1_we_vote_id) and positive_value_exists(politician2_we_vote_id):
+        if politician1_we_vote_id != politician2_we_vote_id:
+            # Conflicting parent politicians
+            # TODO: Call separate politician merge process
+            messages.add_message(request, messages.ERROR, 'Unable to merge politicians at this time.')
+            return HttpResponseRedirect(reverse('candidate:find_and_remove_duplicate_candidates', args=()) +
+                                        "?google_civic_election_id=" + str(google_civic_election_id) +
+                                        "&state_code=" + str(state_code))
+        # else do nothing (same parent politician)
+    elif positive_value_exists(politician2_we_vote_id):
+        # Migrate politician from candidate 2 to candidate 1
+        candidate1_on_stage.politician_we_vote_id = politician2_we_vote_id
+    # else do nothing (no parent politician for candidate 2)
+
+    # TODO: Migrate bookmarks
+
+    # Merge attribute values
+    conflict_values = figure_out_conflict_values(candidate1_on_stage, candidate2_on_stage)
+    for attribute in CANDIDATE_UNIQUE_IDENTIFIERS:
+        conflict_value = conflict_values.get(attribute, None)
+        if conflict_value == "CONFLICT":
+            choice = request.POST.get(attribute + '_choice', '')
+            if candidate2_we_vote_id == choice:
+                setattr(candidate1_on_stage, attribute, getattr(candidate2_on_stage, attribute))
+        elif conflict_value == "CANDIDATE2":
+            setattr(candidate1_on_stage, attribute, getattr(candidate2_on_stage, attribute))
+
+    # Merge public positions counts
+    public_positions_results = move_positions_to_another_candidate(candidate2_on_stage.id, candidate2_we_vote_id,
+                                                                   candidate1_on_stage.id, candidate1_we_vote_id,
```

Comment:
At this point in the code, I would use "candidate1_id" and "candidate1_we_vote_id" in favor of "candidate1_on_stage.id" (as opposed to one value from the incoming variable, and the other from a retrieved object) since the values in the candidate1_on_stage object will be in transition. You will need to create a variable candidate1_id as soon as the candidate1 object is retrieved.